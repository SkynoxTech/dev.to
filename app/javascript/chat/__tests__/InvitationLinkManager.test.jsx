import { h } from 'preact';
import render from 'preact-render-to-json';
import { shallow } from 'preact-render-spy';
import InvitationLinkManager from '../ChatChannelSettings/MembershipManager/InvitationLinkManager';

const data = {
  invitationLink: 'https://dummy.com',
  currentMembership: {
    role: 'mod',
    user_id: 1,
    chat_channel_id: 1,
    id: 2,
    status: 'active',
  },
};

const nonModData = {
  invitationLink: 'https://dummy.com',
  currentMembership: {
    role: 'member',
    user_id: 1,
    chat_channel_id: 1,
    id: 2,
    status: 'active',
  },
};

const getInvitationLinkManager = (resource) => (
  <InvitationLinkManager
    activeMemberships={resource.activeMemberships}
    currentMembership={resource.currentMembership}
  />
);

describe('<getInvitationLinkManager />', () => {
  it('should render the test snapshot', () => {
    const tree = render(getInvitationLinkManager(data));
    expect(tree).toMatchSnapshot();
  });

  it('should have the elements', () => {
    const context = shallow(getInvitationLinkManager(data));

    expect(context.find('.invitation-section').exists()).toEqual(true);
  });

  it('should have the title elements', () => {
    const context = shallow(getInvitationLinkManager(data));

    expect(context.find('.title').exists()).toEqual(true);
  });

  it('should not render elements when user have role member', () => {
    const context = shallow(getInvitationLinkManager(nonModData));

    expect(context.find('.invitation-section').exists()).toEqual(false);
  });

  it('should have the button elements', () => {
    const context = shallow(getInvitationLinkManager(data));

    expect(context.find('.spec__image-markdown-copy').exists()).toEqual(true);
  });

  it('should click on the button', () => {
    const context = shallow(getInvitationLinkManager(data));
    context.find('.spec__image-markdown-copy').simulate('click');
    expect(context.find('#image-markdown-copy-link-announcer').text()).toEqual(
      'Copied!',
    );
  });
});
