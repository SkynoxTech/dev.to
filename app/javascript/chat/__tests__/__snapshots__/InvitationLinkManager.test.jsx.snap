// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<getInvitationLinkManager /> should render the test snapshot 1`] = `
<div
  class="p-4 grid gap-2 crayons-card my-4 invitation-section"
>
  <h3
    class="text-center title"
  >
    Invitation Link
  </h3>
  <clipboard-copy
    aria-controls="image-markdown-copy-link-announcer"
    aria-live="polite"
    class="flex items-center flex-1"
    for="chat-channel-unviation-url"
    onClick={[Function]}
  >
    <input
      class="crayons-textfield mr-2"
      id="chat-channel-unviation-url"
      readOnly="true"
      type="text"
    />
    <button
      class="crayons-btn crayons-btn--ghost crayons-btn--icon spec__image-markdown-copy"
      disabled={false}
      type="button"
    >
      <svg
        aria-labelledby="fc5f15add1e114844f5e"
        class="crayons-icon copy-icon"
        height="24"
        role="img"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="fc5f15add1e114844f5e"
        >
          Copy Markdown for image
        </title>
        <path
          d="M7 6V3a1 1 0 011-1h12a1 1 0 011 1v14a1 1 0 01-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1 1 0 013 21l.003-14c0-.552.45-1 1.007-1H7zm2 0h8v10h2V4H9v2zm-2 5v2h6v-2H7zm0 4v2h6v-2H7z"
        />
      </svg>
    </button>
    <span
      class="fs-s opacity-0"
      id="image-markdown-copy-link-announcer"
      role="alert"
    >
      Copied!
    </span>
  </clipboard-copy>
</div>
`;
